pages/profile/index.js

```diff
@@ -0,0 +1,87 @@
+import IconButton from "@/components/Styled/IconButton";
+import StyledList from "@/components/Styled/StyledList";
+
+import Link from "next/link";
+import styled from "styled-components";
+import Heart from "@/public/icons/heart-svgrepo-com.svg";
+
+import Pot from "@/public/icons/cooking-pot-fill-svgrepo-com.svg";
+import StyledP from "@/components/Styled/StyledP";
+import { useRouter } from "next/router";
+
+export default function ProfilePage() {
+  const router = useRouter();
+  return (
+    <>
+      <IconButton
+        style="Settings"
+        top="var(--gap-out)"
+        left="var(--gap-out)"
+        onClick={() => router.push("/profile/settings")}
+        fill="var(--color-lightgrey)"
+      />
```

Comment:
This looks to me like it could and should be a Link instead of a button.

If you want to keep using the IconButton Component but as a Link component, you can write it like this:

```suggestion
      <IconButton
      as={Link}
        style="Settings"
        top="var(--gap-out)"
        left="var(--gap-out)"
        href="/profile/settings"
        fill="var(--color-lightgrey)"
      />
```

That way it is semantically correct and accepts an href as an attribute.

Reply:
After changing the button to a link, it didn't work as expected. Instead, I modified the router.push() function to router.back() So if the settings page is accessed from elsewhere, the path still works.